src/Arcus.Security.Core/CompositeSecretProvider.cs

```diff
@@ -364,8 +364,8 @@ private void EnsureAnySecretProvidersConfigured(string secretName)
                 _logger.LogSecurityEvent("Get Secret", new Dictionary<string, object>
                 {
                     ["SecretName"] = secretName,
-                    ["SecretProvider"] = source.SecretProvider.GetType().Name
-                }); 
+                    ["SecretProvider"] = source.Options?.Name ?? source.SecretProvider.GetType().Name
```

Comment:
Shouldn't we check for empty string?
```suggestion
                    ["SecretProvider"] = string.IsNullOrWhitespace(source.Options?.Name) ? source.SecretProvider.GetType().Name : source.Options?.Name
```

Reply:
The way we created these options, we already made sure that the model validates this. So, not sure, adding it would temper some paranoia fears but would also indicate that there's may not an validation happening on the options.

Reply:
Ok, fine by me